conformance/tests/basic/inferencepool_multiple_rules_different_pools.yaml

```diff
@@ -0,0 +1,184 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: pool-primary-epp-service
+  namespace: gateway-conformance-app-backend
+spec:
+  selector:
+    app: pool-primary-epp
+  ports:
+  - name: grpc
+    port: 9002
+    targetPort: 9002
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pool-primary-epp
+  namespace: gateway-conformance-app-backend
+  labels:
+    app: pool-primary-epp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: pool-primary-epp
+  template:
+    metadata:
+      labels:
+        app: pool-primary-epp
+    spec:
+      containers:
+      - name: epp
+        # TODO(#996) Switch to use a released version of the image instead of epp:main
+        image: us-central1-docker.pkg.dev/k8s-staging-images/gateway-api-inference-extension/epp:main
+        imagePullPolicy: Always
+        env:
+        - name: ENABLE_REQ_HEADER_BASED_SCHEDULER_FOR_TESTING
+          value: "true"
+        args:
+        - "-poolName"
+        - "pool-primary"
+        - "-poolNamespace"
+        - "gateway-conformance-app-backend"
+        - -grpcPort
+        - "9002"
+        - -grpcHealthPort
+        - "9003"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: pool-secondary-epp-service
+  namespace: gateway-conformance-app-backend
+spec:
+  selector:
+    app: pool-secondary-epp
+  ports:
+  - name: grpc
+    port: 9002
+    targetPort: 9002
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pool-secondary-epp
+  namespace: gateway-conformance-app-backend
+  labels:
+    app: pool-secondary-epp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: pool-secondary-epp
+  template:
+    metadata:
+      labels:
+        app: pool-secondary-epp
+    spec:
+      containers:
+      - name: epp
+        image: us-central1-docker.pkg.dev/k8s-staging-images/gateway-api-inference-extension/epp:main
+        imagePullPolicy: Always
+        env:
+        - name: ENABLE_REQ_HEADER_BASED_SCHEDULER_FOR_TESTING
+          value: "true"
+        args:
+        - "-poolName"
+        - "pool-secondary"
+        - "-poolNamespace"
+        - "gateway-conformance-app-backend"
+        - -grpcPort
+        - "9002"
+        - -grpcHealthPort
+        - "9003"
+---
+apiVersion: networking.gke.io/v1
+kind: HealthCheckPolicy
+metadata:
+  name: primary-pool-health-check
+  namespace: gateway-conformance-app-backend
+spec:
+  targetRef:
+    group: "inference.networking.x-k8s.io"
+    kind: InferencePool
+    name: pool-primary
+  default:
+    config:
+      type: HTTP
+      httpHealthCheck:
+          requestPath: /
+          port: 3000
+---
+apiVersion: networking.gke.io/v1
+kind: HealthCheckPolicy
+metadata:
+  name: secondary-pool-health-check
+  namespace: gateway-conformance-app-backend
+spec:
+  targetRef:
+    group: "inference.networking.x-k8s.io"
+    kind: InferencePool
+    name: pool-secondary
+  default:
+    config:
+      type: HTTP
+      httpHealthCheck:
+          requestPath: /
```

Comment:
HealthCheckPolicy is an implementation-specific resource that should not be included in the conformance tests.

Reply:
Good point, since we are using the header filter to force the EPP behaviour the health-check here is not required. I've removed it and confirmed that the tests pass without it.

Reply:
Done.